### PR TITLE
add intellisence

### PR DESCRIPTION
**Pull request by ceo1647**

### CODE REVIEW

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs [hunk 1]
```diff
@@ -38,5 +40,57 @@ public static void SetCaretToSpecificLineAndColumn(this TextEditor textEditor, i
...
+                    {
+                        continue;
+                    }
+
+                    if (data.Any(x => string.Equals(x.Text, word)))
+                    {
+                        continue;
+                    }
+
+                    if (word.ToLower().StartsWith(autocompletionText.ToLower()))
```
Comment (mkhomutov) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: you probably have to convert `autocompletionText` to lower only once, insted of doing it each time in loop

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs [hunk 2]
```diff
@@ -38,5 +40,57 @@ public static void SetCaretToSpecificLineAndColumn(this TextEditor textEditor, i
             var newCaretOffset = offset + columnOffset;
             textEditor.CaretOffset = maxCaretOffset > newCaretOffset ? newCaretOffset : maxCaretOffset;
         }
+
+        public static List<ICompletionData> GetCompletionDataForText(this TextEditor textEditor, string autocompletionText, int columnIndex, int[][] scheme)
+        {
```
Comment (mkhomutov) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: check `textEditor` to null

File: src/Orc.CsvTextEditor/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -43,6 +44,34 @@ public static string RemoveCommaSeparatedText(this string text, int positionStar
             return text;
         }
 
+        public static string GetWordFromOffset(this string text, int offset)
+        {
+            var textLength = text.Length;
```
Comment (mkhomutov) [inline src/Orc.CsvTextEditor/Extensions/StringExtensions.cs]: what will happened if `text == null`?

File: src/Orc.CsvTextEditor.Tests/StringExtensionsFacts.cs [hunk 4]
```diff
@@ -14,6 +14,17 @@ namespace Orc.CsvTextEditor.Tests
     [TestFixture]
     public class StringExtensionsFacts
     {
+        [TestCase("here, some; wor;ds to test", 0, "here")]
```
Comment (mkhomutov) [inline src/Orc.CsvTextEditor.Tests/StringExtensionsFacts.cs]: probably ';' symbol is typo

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs (hunk 1 above)
Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: yes

File: src/Orc.CsvTextEditor/Extensions/StringExtensions.cs (hunk 3 above)
Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/StringExtensions.cs]: oh yes... need to add arg check

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs (hunk 2 above)
Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: done

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs (hunk 1 above)
Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: done

File: src/Orc.CsvTextEditor/Extensions/StringExtensions.cs (hunk 3 above)
Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/StringExtensions.cs]: done

File: src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs [hunk 5]
```diff
@@ -71,6 +71,8 @@ public static List<ICompletionData> GetCompletionDataForText(this TextEditor tex
                 var columnChunk = text.Substring(columnOffset, columnWidth);
                 var words = columnChunk.Split(null);
                 var currentWord = text.GetWordFromOffset(offset - 1);
+
+                autocompletionText = autocompletionText.ToLower();
```
Comment (mkhomutov) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: we have another loop. The loop of lines and probably we have to move `autocompletionText.ToLower()` out of this loop either

Comment (ceo1647) [inline src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs]: ))

**Pull request merged**